src/pages/dropsite_admin.js

```diff
@@ -1,54 +1,74 @@
 /** @jsx jsx */
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { jsx } from '@emotion/core';
 import { withRouter } from 'react-router-dom';
 
+import DropSiteAdmin from 'components/Dropsite/Admin';
 import Page from 'components/layouts/Page';
-import DropSiteAdmin from 'components/DropSiteAdmin';
-
-class AdminDropSite extends React.Component {
-  componentDidMount() {
-    this.props.backend.getRequests(this.props.match.params.id).then((data) => {
-      this.setState(
-        {
-          needs: data,
-        },
-        () => {},
-      );
-    });
-    this.props.backend.listSupply(this.props.match.params.id).then((data) => {
-      this.setState(
-        {
-          supply: data,
-        },
-        () => {
-          // console.log(this.state);
-        },
-      );
+import { routeWithParams } from 'lib/utils/routes';
+import { Routes } from 'constants/Routes';
+
+function AdminDropSite({ backend, history, match }) {
+  const [dropSite, setDropSite] = useState(undefined);
+  const [requests, setRequests] = useState([]);
+
+  useEffect(() => {
+    backend.getDropSites(match.params.id).then((data) => {
+      data && setDropSite({ ...data, dropSiteId: data.location_id });
     });
-    this.props.backend.getDropSites(this.props.match.params.id).then((data) => {
-      this.setState(
-        {
-          dropSiteId: data?.location_id,
-          dropSiteName: data?.dropSiteName,
-          dropSiteAddress: data?.dropSiteAddress,
-          dropSiteZip: data?.dropSiteZip,
-          dropSiteDescription: data?.dropSiteDescription,
-          dropSiteHospital: data?.dropSiteHospital,
-          dropSitePhone: data?.dropSitePhone,
-        },
-        () => {},
-      );
+    backend.getRequests(match.params.id).then((data) => {
+      data && setRequests(data);
     });
-  }
+  }, [backend, match.params.id]);
+
+  const handleUpdateContact = () => {
+    history.push(
+      routeWithParams(Routes.DROPSITE_CONTACT, {
+        id: dropSite.dropSiteId,
+      }),
+    );
+  };
 
-  render() {
-    return (
-      <Page>
-        <DropSiteAdmin backend={this.props.backend} {...this.state} />
-      </Page>
+  const handleUpdateLocation = () => {
+    history.push(
+      routeWithParams(Routes.DROPSITE_NEW_ADMIN, {
+        id: dropSite.dropSiteId,
+      }),
     );
+  };
+
+  const handleRequestSupplies = () => {
+    history.push(
+      routeWithParams(Routes.SUPPLY_NEW_ADMIN, {
+        id: dropSite.dropSiteId,
+      }),
+    );
+  };
+
+  const deleteRequest = (id) => {
+    backend.deleteRequest(id).then((data) => {
+      setRequests(requests.filter((req) => req.id === data));
+    });
+  };
+
+  if (!dropSite) {
+    return null;
   }
+
+  return (
+    <Page currentProgress={4} totalProgress={5}>
+      <DropSiteAdmin
+        {...{
```

Comment:
Does this need to be spread?

Reply:
it doesn't but i tend to do this with components that have the same keys as props as it doesn't look so redundant versus
```
dropSite={dropSite}
requests={requests}
...
```
and so on if it wasn't spread. for props that have different keys i don't usually put them in this object as I did with onDelete but would move it out as normal props are defined. so I could just pull that one out as `onDelete={deleteRequest}`

do we have a preference? i don't mind either way.